javascript/site-banner.js

```diff
@@ -0,0 +1,44 @@
+(function () {
+    function getStorageKey(id) {
+        return 'SiteBanner_' + id + '_Dismiss';
+    }
+
+    // Called when a user closes a banner.
+    function callback(event) {
+        var button = event.currentTarget;
+        var bannerId = button.dataset.id;
+        var banner = document.getElementById('site-banner-' + bannerId);
+
+        // The banner can only be closed once, so we don't need the click handler anymore.
+        button.removeEventListener('click', callback);
+
+        // Remove the banner from the page.
+        banner.parentNode.removeChild(banner);
+
+        // Make sure the banner doesn't re-appear when the page is re-loaded.
+        sessionStorage.setItem(getStorageKey(bannerId), true);
```

Comment:
Should check for compat, not all browsers have it. Ideally you'd just hide the "close" button if that feature isn't available (e.g. on IE10 and below)

Reply:
have updated, but `sessionStorage` is actually supported down to IE8 :)